### PR TITLE
Various bug fixes in recommenders, minor linting fixes

### PR DESCRIPTION
**Pull request by yrahul3910**
This PR fixes various bugs in the code. Additionally, it adds more style guidelines to the README. Collaborators: please make sure you read and understand these. These are standard practices and *good* practices at that.

### CODE REVIEW

File: src/recommenders/bi.py [hunk 1]
```diff
@@ -9,8 +9,8 @@
 class BIRecommender(BaseRecommender):
     __name__ = 'bayesian inference'
 
-    def __init__(self, stats_df):
-        super().__init__(stats_df)
+    def __init__(self, stats_df, player_name):
```
Comment (yrahul3910) [inline src/recommenders/bi.py]: This needs a type hint.

File: src/recommenders/bi.py [hunk 2]
```diff
@@ -26,27 +26,30 @@ def _recommend(self):
             score = self._objective(config)
             self.scores.append(score)
 
-        for i in range(6):
+        for i in range(2):
```
Comment (yrahul3910) [inline src/recommenders/bi.py]: Incorrect. Fix.

File: src/recommenders/bo.py [hunk 3]
```diff
@@ -16,5 +16,6 @@ def _recommend(self):
             n_initial_points=5,
             verbose=False
         )
-        recommendation = pd.Series(result.x, index=['Character', 'Body', 'Tires', 'Gliders'])
+        recommendation = pd.Series(
```
Comment (yrahul3910) [inline src/recommenders/bo.py:21]: Magic numbers are bad code. Explain in a comment.

File: src/recommenders/bi.py (hunk 2 above)
Comment (yrahul3910) [reply src/recommenders/bi.py]: Actually: these should be constants defined in `BaseRecommender`. Why use magic numbers at all?

File: src/recommenders/bo.py (hunk 3 above)
Comment (yrahul3910) [reply src/recommenders/bo.py:21]: Fixed.

**Review by korchiev — APPROVED** [review]

**Review by vodelya — APPROVED** [review]

**Pull request merged**